Csrc/microphysics_CLIMA.h

```diff
@@ -278,31 +616,37 @@ void CLIMA_entropy_source_formation(const struct DimStruct *dims, struct LookupS
     const ssize_t kmax = dims->nlg[2]-dims->gw;
 
     //entropy tendencies from formation or evaporation of precipitation
-    //we use fact that P = d(qr)/dt > 0, E =  d(qr)/dt < 0
     for(ssize_t i=imin; i<imax; i++){
         const ssize_t ishift = i * istride;
         for(ssize_t j=jmin; j<jmax; j++){
             const ssize_t jshift = j * jstride;
             for(ssize_t k=kmin; k<kmax; k++){
                 const ssize_t ijk = ishift + jshift + k;
 
-                const double lam_T = lam_fp(T[ijk]);
-                const double L_fp_T = L_fp(T[ijk],lam_T);
+                const double lam_T  = lam_fp(T[ijk]);
                 const double lam_Tw = lam_fp(Twet[ijk]);
-                const double L_fp_Tw = L_fp(Twet[ijk],lam_Tw);
-                const double pv_star_T = lookup(LT, T[ijk]);
-                const double pv_star_Tw = lookup(LT,Twet[ijk]);
+
+                const double L_fp_T  = L_fp(T[ijk], lam_T);
+                const double L_fp_Tw = L_fp(Twet[ijk], lam_Tw);
+
+                const double pv_star_T  = lookup(LT, T[ijk]);
+                const double pv_star_Tw = lookup(LT, Twet[ijk]);
+
                 const double pv = pv_c(p0[k], qt[ijk], qv[ijk]);
                 const double pd = p0[k] - pv;
+
                 const double sd_T = sd_c(pd, T[ijk]);
-                const double sv_star_T = sv_c(pv_star_T,T[ijk] );
+
+                const double sv_star_T  = sv_c(pv_star_T,  T[ijk] );
                 const double sv_star_Tw = sv_c(pv_star_Tw, Twet[ijk]);
+
                 const double S_P = sd_T - sv_star_T + L_fp_T/T[ijk];
                 const double S_E = sv_star_Tw - L_fp_Tw/Twet[ijk] - sd_T;
                 const double S_D = -Rv * log(pv/pv_star_T) + cpv * log(T[ijk]/Twet[ijk]);
 
-                entropy_tendency[ijk] += S_P         * 0.5 * (qr_tendency[ijk] + fabs(qr_tendency[ijk])) -
-                                         (S_E + S_D) * 0.5 * (qr_tendency[ijk] - fabs(qr_tendency[ijk]));
+                const double tmp = S_P * precip_formation_rate[ijk] - (S_E + S_D) * evaporation_sublimation_rate[ijk];
```

Comment:
What is this line for?

Reply:
Yeah, we do. Thats a leftover from when I was printing NaNs left and right 😂